src/include/storage/sql_table.h

```diff
@@ -150,21 +293,23 @@ class SqlTable {
    * Generates an ProjectedRowInitializer for the execution layer to use. This performs the translation from col_oid to
    * col_id for the Initializer's constructor so that the execution layer doesn't need to know anything about col_id.
    * @param col_oids set of col_oids to be projected
+   * @param version_num the schema version
    * @return pair of: initializer to create ProjectedRow, and a mapping between col_oid and the offset within the
    * ProjectedRow to create ProjectedColumns, and a mapping between col_oid and the offset within the
    * ProjectedColumn
    * @warning col_oids must be a set (no repeats)
    */
   std::pair<ProjectedRowInitializer, ProjectionMap> InitializerForProjectedRow(
-      const std::vector<catalog::col_oid_t> &col_oids) const {
+      const std::vector<catalog::col_oid_t> &col_oids, layout_version_t version_num) const {
     TERRIER_ASSERT((std::set<catalog::col_oid_t>(col_oids.cbegin(), col_oids.cend())).size() == col_oids.size(),
                    "There should not be any duplicated in the col_ids!");
-    auto col_ids = ColIdsForOids(col_oids);
+    auto col_ids = ColIdsForOids(col_oids, version_num);
     TERRIER_ASSERT(col_ids.size() == col_oids.size(),
                    "Projection should be the same number of columns as requested col_oids.");
+    TERRIER_ASSERT(tables_.Find(version_num) != tables_.CEnd(), "Table version must exist before insert");
```

Comment:
The message seems incorrect. `InitializerForProjectedRow` does not insert any row.